lxd/container_lxc.go

```diff
@@ -1895,19 +1911,36 @@ func (c *containerLXC) startCommon() (string, error) {
 			// Unix device
 			paths, err := c.createUnixDevice(fmt.Sprintf("unix.%s", k), m)
 			if err != nil {
-				return "", err
+				if m["required"] == "" || shared.IsTrue(m["required"]) {
+					return "", err
+				}
+
+				srcPath := m["source"]
+				if srcPath == "" {
+					srcPath = m["path"]
+				}
+				srcPath = shared.HostPath(srcPath)
+
+				err = deviceInotifyAddClosestLivingAncestor(c.state, srcPath)
+				if err != nil {
+					logger.Errorf("Failed to add \"%s\" to inotify targets", srcPath)
+					return "", err
+				}
+				continue
 			}
 			devPath := paths[0]
 			if c.IsPrivileged() && !c.state.OS.RunningInUserNS && c.state.OS.CGroupDevicesController {
 				// Add the new device cgroup rule
 				dType, dMajor, dMinor, err := deviceGetAttributes(devPath)
 				if err != nil {
-					return "", err
-				}
-
-				err = lxcSetConfigItem(c.c, "lxc.cgroup.devices.allow", fmt.Sprintf("%s %d:%d rwm", dType, dMajor, dMinor))
-				if err != nil {
-					return "", fmt.Errorf("Failed to add cgroup rule for device")
+					if shared.IsTrue(m["required"]) {
```

Comment:
Shouldn't that also include m["required"] == ""?

Reply:
added